From 8e29096d70ea1046fc1f76b7cec8fb08af113877 Mon Sep 17 00:00:00 2001
From: Benjamin Tissoires <benjamin.tissoires@redhat.com>
Date: Tue, 26 Aug 2014 17:19:07 -0400
Subject: [input] alps: cache firmware version

Message-id: <1409073549-31548-17-git-send-email-benjamin.tissoires@redhat.com>
Patchwork-id: 89241
O-Subject: [PATCH RHEL7 16/18] Input: alps - cache firmware version
Bugzilla: 1107819
RH-Acked-by: Hans de Goede <hdegoede@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1107819
Upstream: since v3.17
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7882179
Tested by me on a Toshiba Portege R30 with a v7 Alps touchpad

commit c0cd17f6dc7342a81b61017e6b84e363f86081c6
Author: Hans de Goede <hdegoede@redhat.com>
Date:   Fri Jul 25 22:49:14 2014 -0700

    Input: alps - cache firmware version

    Signed-off-by: Hans de Goede <hdegoede@redhat.com>
    Signed-off-by: Dmitry Torokhov <dmitry.torokhov@gmail.com>

Signed-off-by: Benjamin Tissoires <benjamin.tissoires@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/input/mouse/alps.c b/drivers/input/mouse/alps.c
index 8f79491..31b963d 100644
--- a/drivers/input/mouse/alps.c
+++ b/drivers/input/mouse/alps.c
@@ -1901,6 +1901,9 @@ static int alps_identify(struct psmouse *psmouse, struct alps_data *priv)
      alps_exit_command_mode(psmouse))
   return -EIO;
 
+ /* Save the Firmware version */
+ memcpy(priv->fw_ver, ec, 3);
+
  if (alps_match_table(psmouse, priv, e7, ec) == 0) {
   return 0;
  } else if (e7[0] == 0x73 && e7[1] == 0x03 && e7[2] == 0x50 &&
diff --git a/drivers/input/mouse/alps.h b/drivers/input/mouse/alps.h
index 17e3ae3..e3d0f09 100644
--- a/drivers/input/mouse/alps.h
+++ b/drivers/input/mouse/alps.h
@@ -124,6 +124,7 @@ struct alps_fields {
  *   known format for this model.  The first byte of the report, ANDed with
  *   mask0, should match byte0.
  * @mask0: The mask used to check the first byte of the report.
+ * @fw_ver: cached copy of firmware version (EC report)
  * @flags: Additional device capabilities (passthrough port, trackstick, etc.).
  * @x_max: Largest possible X position value.
  * @y_max: Largest possible Y position value.
@@ -149,6 +150,7 @@ struct alps_data {
  int addr_command;
  unsigned char proto_version;
  unsigned char byte0, mask0;
+ unsigned char fw_ver[3];
  int flags;
  int x_max;
  int y_max;
-- 
1.7.1